apps/prairielearn/python/question_phases.py

```diff
@@ -158,20 +158,19 @@ def process_element(
                 return element_value
             elif phase == "file":
                 if result is not None:
-                    raise Exception("Another element already returned a file")
+                    raise RuntimeError("Another element already returned a file")
                 result = element_value
-            else:
-                if element_value is not None and element_value is not data:
-                    # TODO: Once this has been running in production for a while,
-                    # change this to raise an exception.
-                    sys.stderr.write(
-                        f"Function {str(phase)}() in {str(element_controller)} returned a data object other than the one that was passed in.\n\n"
-                        + "There is no need to return a value, as the data object is mutable and can be modified in place.\n\n"
-                        + "For now, the return value will be used instead of the data object that was passed in.\n\n"
-                        + "In the future, returning a different object will trigger a fatal error."
-                    )
-        except Exception as exc:
-            raise Exception(f"Error processing element {element.tag}") from exc
+            elif element_value is not None and element_value is not data:
+                # TODO: Once this has been running in production for a while,
+                # change this to raise an exception.
+                sys.stderr.write(
+                    f"Function {str(phase)}() in {str(element_controller)} returned a data object other than the one that was passed in.\n\n"
+                    + "There is no need to return a value, as the data object is mutable and can be modified in place.\n\n"
+                    + "For now, the return value will be used instead of the data object that was passed in.\n\n"
+                    + "In the future, returning a different object will trigger a fatal error."
+                )
+        except Exception as err:
```

Comment:
👀 For my own understanding, why didn't the linter complain about this?

Reply:
The linter doesn't complain if you use the exception. So If I did `as exc` and used `exc`, it would be happy.